printing/pycode: Min and Max added to _known_functions

AdvaitPote opened the pull request:
<!-- Your title above should be a short description of what
was changed. Do not include the issue number in the title. -->

#### References to other Issues or PRs
<!-- If this pull request fixes an issue, write "Fixes #NNNN" in that exact
format, e.g. "Fixes #1234" (see
https://tinyurl.com/auto-closing for more information). Also, please
write a comment on that issue linking back to this pull request once it is
open. -->

Fixes #22908 

#### Brief description of what is fixed or changed
Code originally property of @ThePauliPrinciple. 
Because of PythonCodePrinter not supporting Min and Max, these functions were added to `_known_functions`.


#### Other comments

Appropriate tests to be added.
#### Release Notes

<!-- Write the release notes for this release below between the BEGIN and END
statements. The basic format is a bulleted list with the name of the subpackage
and the release note for this PR. For example:

* solvers
  * Added a new solver for logarithmic equations.

* functions
  * Fixed a bug with log of integers.

or if no release note(s) should be included use:

NO ENTRY

See https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more
information on how to write release notes. The bot will check your release
notes automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
NO ENTRY
<!-- END RELEASE NOTES -->

AdvaitPote committed:
functions added to _known_functions

praneethratna commented:
The authors test is failing since you have not updated the `.mailmap`. See [Add your name and email address to the .mailmap file](https://github.com/sympy/sympy/wiki/Development-workflow#add-your-name-and-email-address-to-the-mailmap-file) for detailed instructions on how to update the `.mailmap`

oscarbenjamin commented:
You will need to add yourself to the .mailmap file as described here:
https://github.com/sympy/sympy/wiki/Development-workflow#add-your-name-and-email-address-to-the-mailmap-file

AdvaitPote commented:
My name is in the AUTHORS file with my email id and and my first name. Should I change my name from my first name to my full name in the file?

ThePauliPrinciple commented:
This needs tests. Also, please update the title so it is clear what part of sympy this is about, there exist several `_known_functions` variables. e.g. "printing/pycode: Min and Max added to _known_functions".

AdvaitPote commented:
@ThePauliPrinciple I am yet to add tests. Will do in sometime.

About the `.mailmap`, my name and email id is already added in the AUTHORS file as shown:

`Advait <apote2050@gmail.com>`

Is there anything else I am supposed to do?

ThePauliPrinciple commented:
> Is there anything else I am supposed to do?

Yes, you are submitting with your github email, so you need to enter additional information, please read the entire paragraph https://github.com/sympy/sympy/wiki/Development-workflow#add-your-name-and-email-address-to-the-mailmap-file

faze-geek commented:
>I am yet to add tests. Will do in sometime.
> 
The code is dependent on tests and it also gives additional information on what is actually fixed so it's always good to add those .A practice some contributors follow is maybe to mention the status of tests in `Other comments` of the PR description ,in this case being that you will add appropriate tests in a future commit.

AdvaitPote committed:
author: add Advait Pote to .mailmap

AdvaitPote committed:
author: add Advait Pote to .mailmap

AdvaitPote commented:
I have added and committed changes made by me in the `.mailmap`. Is there anything else I am supposed to do because the authors test is still failing. 

Also while running the `mailmap_check.py`, I was continuously getting an error with the error statement as follows:

```
UnicodeDecodeError: 'charmap' codec can't decode byte 0x8d in position 7775: character maps to <undefined>
```

To get rid of this, I replaced `with open(path) as fin:` with `with open(path, encoding='utf8') as fin:` and `with open(path, 'w') as fout:` with `with open(path, 'w', encoding='utf8') as fout:` on lines 288 and 293 respectively. The error was gone after this but after running it, I noticed that the name and email id that I had added wasn't mentioned under 'The following authors will be added to the AUTHORS file at the time of the next SymPy release.'. Despite this, I added and committed these changes as mentioned but changes don't seem to have been reflected. What else needs to be done?

oscarbenjamin commented:
You haven't pushed the changes to .mailmap to GitHub

AdvaitPote committed:
Tests Added for Min and Max

AdvaitPote commented:
Thank you @ThePauliPrinciple @oscarbenjamin  @praneethratna  @faze-geek for your help with this!
